packages/nimble-components/src/rich-text-editor/styles.ts

```diff
@@ -135,17 +171,53 @@ export const styles = css`
         margin-block-end: 0;
     }
 
+    :host([fit-to-content]) .ProseMirror {
+        max-height: fit-content;
+    }
+
     li > p {
         margin-block: 0;
     }
 
+    .ProseMirror p.is-editor-empty:first-child::before {
+        color: ${controlLabelFontColor};
+        content: attr(data-placeholder);
+        float: left;
+        height: 0;
+        pointer-events: none;
+        word-break: break-all;
+    }
+
+    :host([disabled]) .ProseMirror p.is-editor-empty:first-child::before {
+        color: ${controlLabelDisabledFontColor};
+    }
+
     .footer-section {
         display: flex;
         justify-content: space-between;
         border: ${borderWidth} solid transparent;
         border-top-color: rgba(${borderRgbPartialColor}, 0.1);
         height: var(--ni-private-rich-text-editor-footer-section-height);
         overflow: hidden;
+        visibility: visible;
+    }
+
+    :host([footer-hidden]) .footer-section {
+        visibility: hidden;
+        height: 0px;
```

Comment:
Yes, we can set `display: none;` instead of visibility and height. However, while doing so, we have to also add the border width (top + bottom = 2px) of the footer section to the `.ProseMirror` div to hide the footer without any layout shift in the UI as mentioned in the [other comment](https://github.com/ni/nimble/pull/1446#discussion_r1302893177).

Reply:
I think you could avoid that by setting the footer's `box-sizing: border-box` so that the height includes the border, but adding the extra border height seems fine, too.